nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/FetchSolr.java

```diff
@@ -0,0 +1,401 @@
+        relationships.add(ORIGINAL);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    public static final Set<String> SUPPORTED_SEARCH_COMPONENTS = new HashSet<String>();
+    static {
+        SUPPORTED_SEARCH_COMPONENTS.addAll(Arrays.asList(StatsParams.STATS, FacetParams.FACET));
+    }
+    public static final Set<String> SEARCH_COMPONENTS_ON = new HashSet<String>();
+    static {
+        SEARCH_COMPONENTS_ON.addAll(Arrays.asList("true", "on", "yes"));
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final ComponentLog logger = getLogger();
+
+        FlowFile flowFileOriginal = session.get();
+
+        if (flowFileOriginal == null) {
+            if (context.hasNonLoopConnection())
+                return;
+            flowFileOriginal = session.create();
+        }
+
+        final Map<String,String[]> solrParams = parseSolrParams(context, flowFileOriginal);
+        final Set<String> searchComponents = extractSearchComponents(solrParams);
+        final SolrQuery solrQuery = new SolrQuery();
+        solrQuery.add(new MultiMapSolrParams(solrParams));
+
+        solrQuery.setRequestHandler(context.getProperty(REQUEST_HANDLER).getValue());
+
+        final QueryRequest req = new QueryRequest(solrQuery);
+
+        if (isBasicAuthEnabled()) {
+            req.setBasicAuthCredentials(getUsername(), getPassword());
+        }
+        try {
+            final QueryResponse response = req.process(getSolrClient());
+            Map<String,String> attributes = new HashMap<>();
+            attributes.put(ATTRIBUTE_CURSOR_MARK, response.getNextCursorMark());
+            attributes.put(ATTRIBUTE_SOLR_STATUS, String.valueOf(response.getStatus()));
+            attributes.put(ATTRIBUTE_QUERY_TIME, String.valueOf(response.getQTime()));
+
+            if (response.getResults().size() > 0) {
+                FlowFile flowFileResponse = session.create(flowFileOriginal);
+                if (context.getProperty(RETURN_TYPE).getValue().equals(MODE_XML.getValue())){
+                    flowFileResponse = session.write(flowFileResponse, SolrUtils.getOutputStreamCallbackToTransformSolrResponseToXml(response));
+                    flowFileResponse = session.putAttribute(flowFileResponse, CoreAttributes.MIME_TYPE.key(), "application/xml");
+                } else {
+                    final RecordSetWriterFactory writerFactory = context.getProperty(RECORD_WRITER).asControllerService(RecordSetWriterFactory.class);
+                    final RecordSchema schema = writerFactory.getSchema(flowFileOriginal.getAttributes(), null);
+                    final RecordSet recordSet = SolrUtils.solrDocumentsToRecordSet(response.getResults(), schema);
+                    final StringBuffer mimeType = new StringBuffer();
+                    flowFileResponse = session.write(flowFileResponse, new OutputStreamCallback() {

```

Comment:
Compressing this down to a lambda would save a few lines. Not necessary, but your IDE should be able to do it automatically.

Reply:
I will change this and add the curly brackets :)